popover style for apps and user dropdown menus

jancborchardt opened the pull request:
An alternative version of https://github.com/owncloud/core/pull/14731 which doesn’t require us to change all app icons to dark versions.

What do you think @owncloud/designers @karlitschek?

@PVince81 I removed the animation and menuspeed because the jQuery slide function caused the menu to slide but the arrow triangle to directly appear. If we can fix that then we should add a slight animation back at least.

jancborchardt committed:
popover style for apps and user dropdown menus

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/fbfa5c6e-e078-4957-b54d-dc3bc22bf7bf) was created.

jancborchardt commented:
Before:
![dropdown-before](https://cloud.githubusercontent.com/assets/925062/6526430/153e9f8c-c412-11e4-8078-b51d41008d0e.png)
After:
![dropdown-new](https://cloud.githubusercontent.com/assets/925062/6687602/bdfe6e0c-ccac-11e4-977d-947c750fe725.png)

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/fbfa5c6e-e078-4957-b54d-dc3bc22bf7bf) completed: **No new issues**

MorrisJobke commented:
There is a little glitch in the triangle:

![dropdown](https://cloud.githubusercontent.com/assets/245432/6691477/1410ddd8-ccc7-11e4-88f7-bf81d12b8bdc.png)


Also the centered triangle looks weird for short names. Is it possible to move it to the right for the right menu?

MorrisJobke commented:
This is on current Firefoy

jancborchardt commented:
The reason I didn’t move the triangle to the right is that it looks off when it’s not flush with the downward pointing triangle. But to make that flush, it would have to be shifted to the right so far that it looks ridiculous. But yeah, like that it also looks weird.

Can you debug the triangle glitch you get there? Doesn’t happen for me for some reason.

MorrisJobke commented:
It is related to the zoomlevel. I guess we can't do anything here. It's simply a rendering issue of firefox :(

karlitschek commented:
definitely a step into the right direction in my opinion. :+1: Wan we make the dark grey background a bit lighter or darker so that it doesn´t "conflict" too much with the dark blue in the header?  Too bad that we can´t fix the firefox bug.

DeepDiver1975 commented:
@jancborchardt your js change breaks the unit test:

````
Expected true to equal false.
Error: Expected true to equal false.
    at /var/jenkins/workspace/pull-request-analyser-ng-simple/label/SLAVE/core/js/tests/specs/coreSpec.js:443

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/44747c54-2aec-4b43-86be-102c971c0fa0) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/6769986e-04e3-462b-8429-233621626e51) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/30e0455d-5fff-4088-8df5-0454ef98b329) was created.

jancborchardt commented:
@MorrisJobke I moved the triangle of the user menu to the left a bit. Is that better than before?

@karlitschek I made the background darker. :)

@DeepDiver1975 should be fixed now. I accidentally used `show` instead of `hide` in line 480 of js.js

jancborchardt commented:
@owncloud-bot retest this please

MorrisJobke commented:
Looks far better now :)

I tested it with Firefox, Chrome and IE9: Works nice :)

But in IE8 at least the background needs to be added(left menu is okay so far - right menu is completely unreadable
![bildschirmfoto von 2015-03-19 17-42-01](https://cloud.githubusercontent.com/assets/245432/6735620/9a0e07c8-ce5f-11e4-846c-27790476b3d8.png)
):

MorrisJobke commented:
Beside the IE8 bug: :+1:

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/30e0455d-5fff-4088-8df5-0454ef98b329) completed: **No issues found**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/30e0455d-5fff-4088-8df5-0454ef98b329) completed: **No issues found**

karlitschek commented:
Hmmm. I´m not really sure. Looks O.K. but not super awesome.

karlitschek commented:
@owncloud/designers Does someone has a better idea?

jancborchardt commented:
@karlitschek another option would be to make the blue in the header lighter. It's almost black anyway ...

jancborchardt committed:
further adjustments of header and dropdown styles

jancborchardt committed:
more small fixes in header

jancborchardt committed:
adjust share dropdown border radius to 3px

MorrisJobke commented:
@jancborchardt rebase needed

jancborchardt committed:
reduce logo size a bit for a better look

jancborchardt committed:
move triangle of apps dropdown to align with app name triangle

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/e693ccbc-e721-432b-8f67-00fb7fa07664) was created.

jancborchardt commented:
Rebased and also added more fixes (logo smaller, put triangle in right place). Please review again @karlitschek @owncloud/designers

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/e693ccbc-e721-432b-8f67-00fb7fa07664) completed: **No new issues**

karlitschek commented:
looks good in general. :+1:  The arrow in the apps dropdown is not correctly aligned for me :-)

jancborchardt committed:
fix accessibility of ownCloud logo and navigation entries, fix #15013g

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/3e214ea4-464c-43e6-88e5-726fa6eb68b7) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/16055ef5-228a-4a24-9a11-d398cf5d139d) was created.

jancborchardt commented:
@Volker-E in this branch I also fixed the accesibility of the ownCloud logo https://github.com/owncloud/core/issues/15013. As well as the navigation menu – previously it was only tabbed into after tabbing over the search field and the user name.

karlitschek commented:
one more problem. the login screen is also broken

jancborchardt committed:
fix log in issue, fix arrow position, make dropdown work in Firefox

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/8fdc222f-564a-402b-b088-f26760bf138b) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/bff2b081-8c61-4034-8a91-fe32a875c13f) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/90197997-efa7-40d0-b7db-990a951ac096) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/3e214ea4-464c-43e6-88e5-726fa6eb68b7) completed: **No new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/16055ef5-228a-4a24-9a11-d398cf5d139d) completed: **No new issues**

jancborchardt committed:
reintroduce quick animation for menus

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/5eda0353-769e-4dae-a834-2f2377cbdd33) was created.

jancborchardt committed:
adjust animations of Share, Versions and others to OC.menuSpeed

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/a38fb5a7-fbac-4ea1-a089-688319569e6b) was created.

jancborchardt commented:
@karlitschek log in fixed, arrow alignment should be fixed too. Added proper animations.

I also added the same animations for the Versions and Sharing dropdown but for some reason they only work for the _closing_ animation of the Share dropdown. cc @schiesbn – something that we can fix later though.

karlitschek commented:
Arrow alignment is still off :-/  
sharing opening animation doesn´t work as mentioned. maybe something for later or someone else can have a look?
Also the new file dropdown doesn´t seems to be animated. Not sure if this is intended.

jancborchardt commented:
@karlitschek ok, I will come over later to check. About the new file dropdown: Having the animation there caused it to look strange, I can show you later as well.
And the sharing opening animation – yes, something for another pull request since it’s too unrelated to this one.

MorrisJobke commented:
We (Tom, Robin, Joas, Frank) here all agree that this is a nice improvement :) Thanks Jan :)

:+1:

MorrisJobke commented:
* [ ] Have a look at IE8: the personal menu is without background

jancborchardt committed:
fix user dropdown for IE8

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/98d36d0d-0b07-4285-b6d4-d169d6d09834) was created.

jancborchardt committed:
detail corrections for search box layout

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/6a229583-6408-4408-9bc3-16f405890fd1) was created.

MorrisJobke commented:
@tomneedham Please review :)

MorrisJobke commented:
Here is the before and after view: (cc @jospoortvliet for your report ;))

![menu-old](https://cloud.githubusercontent.com/assets/245432/6847895/1aa2cb66-d3ca-11e4-97c4-c6258a14108c.png)
![menu-new](https://cloud.githubusercontent.com/assets/245432/6847897/1c307af0-d3ca-11e4-9aaf-f6d587046eea.png)

MorrisJobke commented:
Jenkins PR: #15221

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/98d36d0d-0b07-4285-b6d4-d169d6d09834) completed: **No new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/6a229583-6408-4408-9bc3-16f405890fd1) completed: **No issues found**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/6a229583-6408-4408-9bc3-16f405890fd1) completed: **No issues found**

jancborchardt committed:
only move dropdown triangle when navigation is opened, fix timing bug

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/d9de3b3c-cc30-458d-928e-02d85b985ee5) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/830e18c0-dd35-4d4d-80e1-ef7f0bfa5cf7) was created.

jancborchardt committed:
mock up missing elements in the tests

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/4cc9f931-ea08-49e7-aab6-f0e7abc2f64a) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/df69dc06-9824-4a63-b994-45dda8fc37a6) was created.

jancborchardt commented:
Test passed via https://github.com/owncloud/core/pull/15233
https://ci.owncloud.org//job/pull-request-analyser-ng-simple/10889/

MorrisJobke commented on core/js/js.js:
```diff
@@ -1196,6 +1196,20 @@ function initCore() {
+	// move triangle of apps dropdown to align with app name triangle
+	// 2 is the additional offset between the triangles
+	if($('#navigation').length) {
+		$('#header #owncloud + .menutoggle').one('click', function(){
```
@jancborchardt Shouldn't the `one()` be an `on()`?

MorrisJobke replied:
Got it... it's a one time event. Nevermind

jancborchardt replied:
Yup ;) sorcery learnt from our resident wizard @PVince81 :star2:
